pkg/evaluators/authorization/opa.go

```diff
@@ -77,10 +77,13 @@ type OPA struct {
 	policyName string
 	policyUID  string
 
-	mu sync.Mutex
+	mu sync.RWMutex
```

Comment:
I might be misreading this, but I think `precompilePolicy` seem to mutate `opa *OPA.opa`… and read `opa *OPA.AllValues`… I have not dug much deeper than looking at what read/writes to the data structure…

Reply:
`precompilePolicy()` is only called by `updateRego(...)`; the latter is acquiring the lock.